src/Nokia_LCD.cpp

```diff
@@ -263,11 +263,27 @@ bool Nokia_LCD::print(unsigned long number) {
     return print(number_as_string);
 }
 
+
 bool Nokia_LCD::print(double number, unsigned short decimals) {
     double integral = 0;
     long fractional = pow(10.0, decimals) * modf(number, &integral);
+    bool out_of_bounds = false;
 
-    return print(static_cast<long>(number)) || print(".") || print(fractional);
+    //prints the left hand side of the dot
+    out_of_bounds = out_of_bounds || print(static_cast<long>(number));
+    out_of_bounds = out_of_bounds || print(".");
+
+    //prints any leading 0s after the dot
+    for(int d = decimals - 1 ; out_of_bounds || (0 <= d); d-- ) {
+        int num = fractional/pow(10, d);
+        if (num == 0) 
+ 	   out_of_bounds = out_of_bounds || print(num);
+	else
+           break; //no more 0s to print so break
+    }
```

Comment:
This is technically correct and saves us a couple of loops if things have gone out of bounds.
However, wasting a few loops in the name of simplicity is I believe preferred in this occasion. The code would now look like this:

```cpp
    for (int d = decimals - 1; d >= 0; d--) {
        int num = fractional / pow(10, d);
        if (num == 0) {
            out_of_bounds = out_of_bounds || print(num);
        } else {
            break;
       }
    }
```

I also added some braces after the `if-else` statement and formatted the spaces to be consistent with the rest of the code.

Reply:
Ok. Done.